src/docs/self-hosted/csp.mdx

```diff
@@ -0,0 +1,26 @@
+---
+title: "Self-Hosted Content Security Policy (CSP)"
+---
+
+<Alert level="warning">
+  This is an experimental feature. Use it with caution.
+</Alert>
+
+Starting with Sentry `23.5.0`, it is possible to enable [CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) on the self-hosted Sentry installation. The good news is that Sentry itself supports [CSP reports collection](https://docs.sentry.io/product/security-policy-reporting/). We recommend creating a separate Sentry project for CSP reports. To enable CSP and reports collection, you'll want to configure the following settings in `sentry.conf.py`:
+
+```
+if "csp.middleware.CSPMiddleware" not in MIDDLEWARE:
+    MIDDLEWARE = ("csp.middleware.CSPMiddleware",) + MIDDLEWARE
+CSP_REPORT_URI = "https://your-sentry-url-prefix.com/api/{csp-project-id}/security/?sentry_key={sentry-key}"
+CSP_REPORT_ONLY = True
+```
```

Comment:
```suggestion
```python
if "csp.middleware.CSPMiddleware" not in MIDDLEWARE:
    MIDDLEWARE = ("csp.middleware.CSPMiddleware",) + MIDDLEWARE
CSP_REPORT_URI = "https://your-sentry-url-prefix.com/api/{csp-project-id}/security/?sentry_key={sentry-key}"
CSP_REPORT_ONLY = True
```
```